engine/compiler/compiler.go

```diff
@@ -497,11 +497,7 @@ func (c *Compiler) Compile(ctx context.Context, args runtime.CompilerArgs) runti
 		}
 
 		for _, s := range step.Secrets {
-			// if the secret was already fetched and stored in the
-			// secret map it can be skipped.
-			if _, ok := spec.Secrets[s.Name]; ok {
```

Comment:
why do we need to remove this? What's causing the issue - is that spec.Secrets has a secret but it doesn't get added to step.SpecSecrets? If so, we can just make sure that we append here - it saves on us having to find the secret each time even if it's already been added.